### DISCUSSION TITLE
Question regarding searching with @@unique constraint

### DISCUSSION THREAD

**Discussion by Yzma** [prisma/prisma#6507] [Q&A] [answered]
Hello! I'm trying to delete a Project using the modal below:

```prisma
model Project {
  id           Int      @id @default(autoincrement())
  name         String
  description  String?
  private      Boolean
  creationDate DateTime @default(now()) @map("creation_date")
  lastUpdated  DateTime @default(now()) @map("last_updated")

  ownerUser    User     @relation(fields: [ownerUserID], references: [id])
  ownerUserID  Int      @map("owner_user_id")

  labels       Label[]
  tickets      Ticket[]

  @@unique([name, ownerUserID])
  @@map("projects")
}

model User {
  id            Int       @id @default(autoincrement())
  name          String?   @unique
  email         String?   @unique
  emailVerified DateTime? @map("email_verified")
  image         String?
  createdAt     DateTime  @default(now()) @map("created_at")
  updatedAt     DateTime  @default(now()) @map("updated_at")

  bio           String?

  organizations Organization[]
  projects      Project[]
  tickets       Ticket[]
  comments      Comment[]

  @@map("users")
}
```

And this is what I tried:

```ts
  const deleteUserProjectResult = await prisma.project.delete({
      where: {
          name_ownerUserID: {
              name: projectName,
              ownerUser:  {
                name: username
            }
          }
      }
  })
```

Essentially I need to specify a unique attribute with a related field. I don't know the ownerUserID in advance, I only know the name (which is username). I tried looking at the documentation (https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#defining-an-id-field) but there isn't an example on how to search/delete using relation fields. Any help is appreciated, thanks!

**Comment by ryands17** [comment, marked as answer]
@Yzma 👋 

For multiple `@@unique` fields, you need all the fields to delete a specific item. If you just have the `username`, you can use `deleteMany`, but that will delete all the projects that the user has. If you do not want that behaviour, then you need at least the `id` of the project or both the unique values to just delete that project.

Hope that makes sense :)

**Reply by Yzma** [reply]
Makes sense, I had a feeling it would come to that when I look at it more. Looks like I'll be taking the deleteMany route then. Appreciate the reply!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
